official/resnet/resnet_run_loop.py

```diff
@@ -644,3 +655,11 @@ def define_resnet_flags(resnet_size_choices=None):
     flags.DEFINE_string(**choice_kwargs)
   else:
     flags.DEFINE_enum(enum_values=resnet_size_choices, **choice_kwargs)
+
+
+def resolve_flag_conflict():
+  if (flags.FLAGS.use_synthetic_data and
```

Comment:
I don't understand how this change is still backward compatible.. you're still requiring users to provide `data_source_type` even if they provided `use_synthetic_data` ? i think this should throw an error only if both are present and do not match. 
And we should have synthetic behavior if `use_synthetic_data` is set, even if `data_source_type` is not set.

Reply:
I think the reason this happens is because I set the data_source_type default to real_data. I now set it to None because whatever I set the default value to be, there will be such problem.